experimental/packages/opentelemetry-sdk-metrics-base/src/view/AttributesProcessor.ts

```diff
@@ -23,15 +23,22 @@ import { Attributes } from '@opentelemetry/api-metrics-wip';
  * additional dimension(s) from the {@link Context}.
  */
 export abstract class AttributesProcessor {
-  abstract process(incoming: Attributes, context: Context): Attributes;
+  /**
+   * Process the metric instrument attributes.
+   *
+   * @param incoming The metric instrument attributes.
+   * @param context The active context when the instrument is synchronous.
+   * `undefined` otherwise.
+   */
+  abstract process(incoming: Attributes, context?: Context): Attributes;
 
   static Noop() {
     return NOOP;
   }
 }
 
 export class NoopAttributesProcessor extends AttributesProcessor {
-  process(incoming: Attributes, _context: Context) {
+  process(incoming: Attributes, _context?: Context) {
```

Comment:
in which case can the context be undefined ? we always have the root context i believe

Reply:
It is determined by the spec that:

> ... Measurements recorded by synchronous instruments can be associated with the Context.
> ... Measurements recorded by asynchronous instruments cannot be associated with the Context.
> https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument

So it will be undefined when the measurement is measured with asynchronous instruments.

Reply:
It is also documented here https://github.com/open-telemetry/opentelemetry-js/pull/2686/files#diff-b0a6ea83e6ececc0eb590febf03e6f8ecd08be484ed0f19ddc3404fb7c77c8c4R30.